src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> top;
+    private Node<T> bottom;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            linkTop(value);
+        } else {
+            linkBottom(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == 0 || size == 0) {
+            linkTop(value);
+        } else if (index == size) {
+            linkBottom(value);
+        } else {
+            Node<T> k = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(k.prev, value, k);
+            k.prev.next = newNode;
+            k.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (T item : list) {
+                add(item);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkItemIndex(index);
+        return getNodeByIndex(index).item;
```

Comment:
`getNodeByIndex()` method should contain `checkItemIndex()` instead of line by line call this method. Fix it everywhere
```suggestion
        return getNodeByIndex(index).item;
```